testdata/go/v3/monitoring/memcached-operator/controllers/memcached_controller.go

```diff
@@ -36,11 +37,15 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+
 	cachev1alpha1 "github.com/example/memcached-operator/api/v1alpha1"
 	"github.com/example/memcached-operator/monitoring"
 )
 
 const memcachedFinalizer = "cache.example.com/finalizer"
+const ruleName = "memcached-operator-rules"
+const namespace = "memcached-operator-system"
```

Comment:
can we get this from `req.NamespacedName`? This contains the name and namespace of the object that would be reconciled. Should the `monitoringv1.PrometheusRule` be in the same location?

Reply:
I wasn't able to get this from `req.NamespacedName` which AFAIK refers to `memcached` here. 
Yes, we create the `monitoringv1.PrometheusRule` in the same namespace.